homeassistant/components/binary_sensor/octoprint.py

```diff
@@ -35,8 +35,11 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the available OctoPrint binary sensors."""
-    octoprint_api = hass.data[DOMAIN]["api"]
     name = config.get(CONF_NAME)
+    if name not in hass.data[DOMAIN]:
+        raise vol.Invalid('Please configure the octoprint component for this '
```

Comment:
It would be better to load this platform from the component with discovery_info to avoid having the user juggle multiple configs.

Reply:
I added automatic loading for the sensor and binary_sensor platform, it's indeed a good idea to load them automatically without having to make the user add more configuration.

I ran into an issue with this, unfortunately. If the user has a configuration section already defined, it'll try to load the component and it'll fail. Considering the current users of this component already have one, this is annoying.

I would like to either ignore the configuration of the sensor/binary_sensor if they are present or, preferably, make it possible to override the components configuration.

It seems to create completely different instances of the components, one auto loaded with load_platform and one from the configuration.

Reply:
Done